content/6_Advanced/Wavelet.mdx

```diff
@@ -32,9 +30,339 @@ Like a segment tree on values rather than indices.
 	</Resource>
 </Resources>
 
+## Introduction
+Suppose you want to support the following queries:
+
+- Given a range, count the number of occurrences of value $x$.
+- Given a range, find the $k$ smallest element
+
+With a wavelet tree, you can easily support these queries in $\mathcal{O}(\log M)$ time,
+where $M$ is the maximum value in the array.
```

Comment:
ok so this aligns with the focus problem right
if so you can just delete this, we assume ppl read the problem beforehand